components/utils.py

```diff
@@ -705,65 +714,75 @@ class CtxPropertyFactory(object):
                             'use_existing_on_upgrade']
 
     # Create node properties according to the workflow context install/upgrade
-    def create(self, service_name, write_to_file=True):
+    def create(self, service_name):
         """
         a Factory used to create a local copy of the node properties used
         upon deployment. This copy will allows to later reuse the properties
         for upgrade/rollback purposes. The node ctx properties will be set
         according to the node property named 'use_existing_on_upgrade'.
         :param service_name: The service name
-        :param write_to_file: write props to local file path
         :return: The relevant ctx node properties dict.
         """
-        ctx_props = self._load_ctx_properties(service_name)
-
         if is_upgrade:
             # archive path existence will determine whether rollback is even
             # necessary if these scripts run in rollback context.
-            archive_path = self._get_rollback_properties_dir(service_name)
-            mkdir(archive_path)
+            self._archive_properties(service_name)
+
+        # write properties to a local backup path
+        ctx_props = self._load_ctx_properties(service_name)
 
-        if write_to_file:
-            # write properties to a local backup path
-            self._write_props_to_file(ctx_props, service_name)
+        self._write_props_to_file(ctx_props, service_name)
 
         return ctx_props
 
-    def _write_props_to_file(self, ctx_props, service_name):
+    def get(self, service_name, upgrade_props=True):
+        """
+        get node properties by service name.
```

Comment:
Please keep the format of the docstrings

capital letter, first line in the same line where the triple double-quotes are, line break between the explanation and the params..